discovery-provider/src/queries/get_tracks.py

```diff
@@ -1,14 +1,78 @@
 import logging # pylint: disable=C0302
 
+from sqlalchemy import func
+from sqlalchemy.sql.functions import coalesce
 from src.models import AggregatePlays, Track, User
 from src.utils import helpers
 from src.utils.db_session import get_db_read_replica
-from src.queries.query_helpers import paginate_query, parse_sort_param, \
+from src.queries.query_helpers import add_query_pagination, get_pagination_vars, parse_sort_param, \
   populate_track_metadata, get_users_ids, get_users_by_id
 from src.queries.get_unpopulated_tracks import get_unpopulated_tracks
 
 logger = logging.getLogger(__name__)
 
+def _get_tracks(session, args):
+    # Create initial query
+    base_query = session.query(Track)
+    base_query = base_query.filter(Track.is_current == True, Track.is_unlisted == False, Track.stem_of == None)
+
+    # Conditionally process an array of tracks
+    if "id" in args:
+        track_id_list = args.get("id")
+        try:
+            # Update query with track_id list
+            base_query = base_query.filter(Track.track_id.in_(track_id_list))
+        except ValueError as e:
+            logger.error("Invalid value found in track id list", exc_info=True)
+            raise e
+
+    # Allow filtering of tracks by a certain creator
+    if "user_id" in args:
+        user_id = args.get("user_id")
+        base_query = base_query.filter(
+            Track.owner_id == user_id
+        )
+
+    # Allow filtering of deletes
+    if "filter_deleted" in args:
+        filter_deleted = args.get("filter_deleted")
+        if filter_deleted:
+            base_query = base_query.filter(
+                Track.is_delete == False
+            )
+
+    if "min_block_number" in args:
+        min_block_number = args.get("min_block_number")
+        base_query = base_query.filter(
+            Track.blocknumber >= min_block_number
+        )
+
+    if "sort" in args:
+        if args["sort"] == "date":
+            base_query = base_query.order_by(
+                coalesce(
+                    func.to_date_safe(Track.release_date, 'Dy Mon DD YYYY HH24:MI:SS'),
```

Comment:
does this intentionally do to_date without the timestamp? the output would be day with secondary sort of id within the same day

Reply:
We actually only support date granularity in the date picker so it doesn't matter

Reply:
Could you also just add a comment here saying this is a custom function defined in the migrations